classes/tests/test_events.py

```diff
@@ -62,27 +81,31 @@ def test_scooter_arrival(self):
         self.assertEqual(self.world.time, self.departure_time + self.travel_time)
 
         # test if arrival cluster contains the arrived scooter
-        self.assertTrue(arrival_cluster.scooters.__contains__(scooter))
+        self.assertIn(scooter, arrival_cluster.scooters)
 
         # test if battery has decreased
         self.assertLess(scooter.battery, scooter_battery)
 
     def test_vehicle_arrival(self):
-        random_cluster_in_state = self.world.state.get_random_cluster(
-            exclude=self.world.state.current_cluster
+        # Choose a random cluster for the vehicle to be in
+        arrival_cluster = self.large_world.state.get_random_cluster(
+            exclude=self.vehicle_large_world.current_location
         )
+        self.vehicle_large_world.current_location = arrival_cluster
         # Create a vehicle arrival event with a arrival time of 20 arriving at a random cluster in the world state
-        vehicle_arrival = VehicleArrival(20, random_cluster_in_state.id)
+        vehicle_arrival_event = VehicleArrival(
+            20, arrival_cluster.id, self.vehicle_large_world
+        )
 
         # Perform the vehicle arrival event
-        vehicle_arrival.perform(self.large_world)
+        vehicle_arrival_event.perform(self.large_world)
 
         # test if the time of world object is set to the departure time
-        self.assertEqual(vehicle_arrival.time, self.large_world.time)
+        self.assertEqual(vehicle_arrival_event.time, self.large_world.time)
 
-        # New current cluster is not the arrival cluster, as the action takes the state to a new cluster
+        # New current cluster is not the arrival cluster, as the do_action takes the vehicle to a new cluster
         self.assertNotEqual(
-            random_cluster_in_state.id, self.large_world.state.current_cluster.id
+            arrival_cluster.id, self.large_world.state.current_cluster.id
```

Comment:
This test will fail since the state doesn't have current_cluster/location. Shold be ```arrival_cluster.id,vehicle.current_location.id ``` if line 94 above is removed